be/src/storage/lake/lake_persistent_index.cpp

```diff
@@ -0,0 +1,294 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "storage/lake/lake_persistent_index.h"
+
+#include "gen_cpp/persistent_index.pb.h"
+#include "storage/chunk_helper.h"
+#include "storage/lake/meta_file.h"
+#include "storage/primary_key_encoder.h"
+#include "storage/tablet_meta_manager.h"
+
+namespace starrocks {
+namespace lake {
+
+Status LakePersistentIndex::load_from_lake_tablet(starrocks::lake::Tablet* tablet, const TabletMetadata& metadata,
+                                                  int64_t base_version, const MetaFileBuilder* builder) {
+    MonotonicStopWatch timer;
+    timer.start();
+
+    if (!is_primary_key(metadata)) {
+        LOG(WARNING) << "tablet: " << tablet->id() << " is not primary key tablet";
+        return Status::NotSupported("Only PrimaryKey table is supported to use persistent index");
+    }
+
+    PersistentIndexMetaPB index_meta;
+
+    // persistent_index_dir has been checked
+    Status status = TabletMetaManager::get_persistent_index_meta(
+            StorageEngine::instance()->get_persistent_index_store(), tablet->id(), &index_meta);
+    if (!status.ok() && !status.is_not_found()) {
```

Comment:
if status is not found error, what will happen?

Reply:
It will rebuild the pk index in following logic